src/openrct2/paint/Painter.cpp

```diff
@@ -102,8 +102,22 @@ void Painter::PaintReplayNotice(DrawPixelInfo& dpi, const char* text)
     GfxSetDirtyBlocks({ screenCoords, screenCoords + ScreenCoordsXY{ stringWidth, 16 } });
 }
 
+static bool ShouldShowFPS()
+{
+    if (gScreenFlags & SCREEN_FLAGS_TITLE_DEMO && !TitleShouldHideVersionInfo())
```

Comment:
I forgot to ask earlier: what is the reasoning behind this specific `if`?

Reply:
This if statement was needed because in the TitleDemo page there is no TopToolbar window so it would not display the FPS counter. This check ensures that if the widgets are not hidden for then it displays the FPS Counter. I think that I can clean it up though, does this look better?

```cpp
static bool ShouldShowFPS()
{
    return (gScreenFlags & SCREEN_FLAGS_TITLE_DEMO && !TitleShouldHideVersionInfo()) || WindowFindByClass(WindowClass::TopToolbar);
}
```

Reply:
No need to "clean it up", all that does is make it harder to understand, while not having any effect on the generated assembly.